src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py

```diff
@@ -1139,41 +1139,39 @@ def _conv_out_length(input_length, kernel_size, stride):
 
         return input_lengths
 
-    def _mask_hidden_states(
-        self, hidden_states: tf.Tensor, mask_time_indices: Optional[tf.Tensor] = None, training: bool = False
-    ):
+    def _mask_hidden_states(self, hidden_states: tf.Tensor, mask_time_indices: Optional[tf.Tensor] = None):
         """
         Masks extracted features along time axis and/or along feature axis according to `SpecAugment
         <https://arxiv.org/abs/1904.08779>`__ .
         """
+        batch_size, sequence_length, hidden_size = shape_list(hidden_states)
 
         # `config.apply_spec_augment` can set masking to False
         if not getattr(self.config, "apply_spec_augment", True):
             return hidden_states
 
         if mask_time_indices is not None:
             # apply SpecAugment along time axis with given mask_time_indices
-            hidden_states = tf.tensor_scatter_nd_update(hidden_states, mask_time_indices, self.masked_spec_embed)
-        elif self.config.mask_time_prob > 0 and training:
-            # generate indices & apply SpecAugment along time axis
-            batch_size, sequence_length, hidden_size = hidden_states.shape
+            hidden_states = hidden_states * tf.expand_dims(mask_time_indices, -1)
+        elif self.config.mask_time_prob > 0:
 
+            # generate indices & apply SpecAugment along time axis
             mask_time_indices = _compute_mask_indices(
                 (batch_size, sequence_length),
-                self.config.mask_time_prob,
-                self.config.mask_time_length,
+                mask_prob=self.config.mask_time_prob,
+                mask_length=self.config.mask_time_length,
                 min_masks=2,
             )
-            hidden_states = tf.tensor_scatter_nd_update(hidden_states, mask_time_indices, self.masked_spec_embed)
+            hidden_states = hidden_states * tf.expand_dims(mask_time_indices, -1)
```

Comment:
This is wrong. The indices represent with 1's the position to mask, so this essentially zero's out everything not in the mask window. If I still wanted to multiply it I would need to invert the mask.

Reply:
it should be this [op](https://www.tensorflow.org/api_docs/python/tf/tensor_scatter_nd_update), but I'm not doing it correctly. The indices tests pass, so I'm probably not passing the indices in the right format.